From 72560942b068093abea1a35955d126791937f173 Mon Sep 17 00:00:00 2001
From: Robert Chiras <robert.chiras@nxp.com>
Date: Mon, 3 Dec 2018 15:09:04 +0200
Subject: [PATCH 5238/5242] MA-13638-2: arch: arm64: dts: Add display-timins
 node for dcss-rm67191

commit  027f47f2b892e3ac9285207cdc6c7524ec4d895d from
https://source.codeaurora.org/external/imx/linux-imx.git

DCSS needs special display timings for MIPI-DSI panel in order to do a
proper display on it.
In order to not break other display controller using this panel, add
custom display-timings for this specific use-case

Signed-off-by: Robert Chiras <robert.chiras@nxp.com>
Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../dts/freescale/fsl-imx8mq-evk-dcss-rm67191.dtsi |   19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-rm67191.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-rm67191.dtsi
index ef62676..d801dba 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-rm67191.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-rm67191.dtsi
@@ -95,6 +95,25 @@
 		dsi-lanes = <4>;
 		panel-width-mm = <68>;
 		panel-height-mm = <121>;
+
+		display-timings {
+			timing {
+				clock-frequency = <132000000>;
+				hactive = <1080>;
+				vactive = <1920>;
+				hfront-porch = <20>;
+				hsync-len = <2>;
+				hback-porch = <34>;
+				vfront-porch = <10>;
+				vsync-len = <5>;
+				vback-porch = <4>;
+				hsync-active = <0>;
+				vsync-active = <0>;
+				de-active = <0>;
+				pixelclk-active = <0>;
+			};
+		};
+
 		port {
 			panel1_in: endpoint {
 				remote-endpoint = <&mipi_dsi_bridge_out>;
-- 
1.7.9.5

